src/uint/mul_mod.rs

```diff
@@ -64,7 +76,7 @@ impl<const LIMBS: usize> MulMod for Uint<LIMBS> {
     type Output = Self;
 
     fn mul_mod(&self, rhs: &Self, p: &Self) -> Self {
-        self.mul_mod(rhs, p)
+        self.mul_mod_vartime(rhs, &p.to_odd().expect("only odd moduli supported"))
     }
```

Comment:
Perhaps the `MulMod` trait should be changed to accept `&Odd<Self>` for `p`, so that the inherent method matches the trait.

(I do find a type safe `Odd` preferable to panicking here)